src/raft.rs

```diff
@@ -828,6 +828,11 @@ impl<T: Storage> Raft<T> {
         // but doesn't change anything else. In particular it does not increase
         // self.term or change self.vote.
         self.state = StateRole::PreCandidate;
+        // If a network partition happens, and leader is in minority partition,
+        // it will step down, and become follower without notifying others.
+        if self.check_quorum {
```

Comment:
Why? I think it make sense to reset it whenever it starts elections.

Reply:
Ok. :+1: